File: cpp/fast_reactor1g.py
class_ds = \
"""A One-Group Fast Reactor Fuel Cycle Component.  This is a daughter class 
of Reactor1G and a granddaughter of FCComp.

Parameters
----------
lib : str, optional
    The path the the FR HDF5 data library.  This value is set to Reactor1G.libfile 
    and used by Reactor1G.loadlib().
rp : ReactorParameters, optional
    The physical reactor parameter data to initialize this FR instance with.  
    If this argument is not provided, default values are taken.
n : str, optional
    The name of this FR instance.

"""

desc = {
    'docstrings': {
        'class': class_ds,
        'attrs': {},
        'methods': {},
        },
    'attrs': {},
    'extra': {},
    }

mod = {'FastReactor1G': desc,
       'docstring': "Python wrapper for FR1G.",}

desc['docstrings']['methods']['calc_params'] = \
"""Along with its own parameter set to track, the FR model implements its own 
function to set these parameters.  This function is equivalent to the following::

    self.params_prior_calc["BUd"]  = 0.0
    self.params_after_calc["BUd"] = self.BUd

    self.params_prior_calc["TRUCR"]  = 0.0
    self.params_after_calc["TRUCR"] = self.calc_tru_cr()

    self.params_prior_calc["P_NL"]  = 0.0
    self.params_after_calc["P_NL"] = self.P_NL

    self.params_prior_calc["U"]  = self.mat_feed_u.mass
    self.params_after_calc["U"] = self.mat_prod_u.mass

    self.params_prior_calc["TRU"]  = self.mat_feed_tru.mass
    self.params_after_calc["TRU"] = self.mat_prod_tru.mass

    self.params_prior_calc["ACT"]  = self.mat_feed_act.mass
    self.params_after_calc["ACT"] = self.mat_prod_act.mass

    self.params_prior_calc["LAN"]  = self.mat_feed_lan.mass
    self.params_after_calc["LAN"] = self.mat_prod_lan.mass

    self.params_prior_calc["FP"]  = 1.0 - self.mat_feed_act.mass  - self.mat_feed_lan.mass
    self.params_after_calc["FP"] = 1.0 - self.mat_prod_act.mass - self.mat_prod_lan.mass

"""

File: cpp/fast_reactor1g.cpp
// One-Group Fast Reactor Component Class

#include "fast_reactor1g.h"

bright::FastReactor1G::FastReactor1G() : bright::Reactor1G(fr_defaults, fr_p2track)
{
};


bright::FastReactor1G::FastReactor1G(std::string lib, std::string n) : bright::Reactor1G(fr_defaults, fr_p2track, n)
{
  libfile = lib;
  loadlib(lib);
};

bright::FastReactor1G::FastReactor1G(ReactorParameters rp, std::string n) : bright::Reactor1G(rp, fr_p2track, n)
{
};

bright::FastReactor1G::FastReactor1G(std::string lib, ReactorParameters rp, std::string n) : bright::Reactor1G(rp, fr_p2track, n)
{
  libfile = lib;
  loadlib(lib);
};

bright::FastReactor1G::~FastReactor1G() 
{
};


void bright::FastReactor1G::calc_params()
{
  /** Sets relevent FR parameters.
   *  Overwrites standard, do-nothing calc_params() function.
   */

  calc_sub_mats();

  params_prior_calc["BUd"]  = 0.0;
  params_after_calc["BUd"] = BUd;

  params_prior_calc["TRUCR"]  = 0.0;
  params_after_calc["TRUCR"] = calc_tru_cr();

  params_prior_calc["P_NL"]  = 0.0;
  params_after_calc["P_NL"] = P_NL;

  params_prior_calc["U"]  = mat_feed_u.mass;
  params_after_calc["U"] = mat_prod_u.mass;

  params_prior_calc["TRU"]  = mat_feed_tru.mass;
  params_after_calc["TRU"] = mat_prod_tru.mass;

  params_prior_calc["ACT"]  = mat_feed_act.mass;
  params_after_calc["ACT"] = mat_prod_act.mass;

  params_prior_calc["LAN"]  = mat_feed_lan.mass;
  params_after_calc["LAN"] = mat_prod_lan.mass;

  params_prior_calc["FP"]  = 1.0 - mat_feed_act.mass  - mat_feed_lan.mass;
  params_after_calc["FP"] = 1.0 - mat_prod_act.mass - mat_prod_lan.mass;

  return;
};
